File: angular-app/src/app/services/route.service.ts
import { HttpHeaders } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { RouteDTO } from '../dto/route-dto';
import { ApiService } from './api.service';

@Injectable({
  providedIn: 'root'
})

export class RouteService {
  
  private basUrl = "route";

  private httpOptions = {
    headers: new HttpHeaders({
      "Content-Type": "application/json",
    })
  };
  
  constructor(private apiService: ApiService) {}
  
  getRouteList(): Observable<RouteDTO[]> {
    return this.apiService.get(`${this.basUrl}/all`);
  }

  createRoute(routeDTO: RouteDTO): Observable<Object> {
    return this.apiService.post(`${this.basUrl}/create`, routeDTO);
  }

  getRouteById(idRoute: number): Observable<RouteDTO>{
    return this.apiService.get(`${this.basUrl}/read/${idRoute}`);
  }

  updateRoute(idRoute: number, routeDTO: RouteDTO): Observable<Object>{
    return this.apiService.put(`${this.basUrl}/update/${idRoute}`, routeDTO);
  }

  deleteRoute(idRoute:number): Observable<Object>{
    return this.apiService.delete(`${this.basUrl}/delete/${idRoute}`);
  }
  
}

File: TransmiApp/TransmiApp/src/main/java/com/example/TransmiApp/service/RouteService.java
package com.example.TransmiApp.service;

import java.util.List;

import com.example.TransmiApp.dto.RouteDTO;
import com.example.TransmiApp.model.Route;

public interface RouteService {
    List<Route> getAllRoutes();
    List<Long> getRouteIds();
    RouteDTO getRouteById(Long idRoute);
    RouteDTO createRoute(RouteDTO routeDTO);
    RouteDTO updateRoute(Long idRoute, RouteDTO routeDTO);
    void deleteRoute(Long idRoute);
}
